sdk-setup/src/mb2

```diff
@@ -3838,8 +3885,28 @@ done
 
 global_lock_init
 
-mkdir -p "$STATEDIR" || exit
-echo '*' > "$STATEDIR/.gitignore"
+case $1 in
+    build-init | build | qmake | cmake )
+        if [[ $1 == build-init && -d $STATEDIR ]]; then
+            notice "Already initialized. Preserving existing configuration."
+        fi
+        mkdir -p "$STATEDIR" || exit
+        echo '*' > "$STATEDIR/.gitignore"
+        ;;
+    check | run | ssh )
+        # These can deal with STATEDIR missing
+        ;;
+    *)
+        if [[ ! -d $STATEDIR ]]; then
+            if ancestor=$(find_upwards "$PWD" ".$WE"); then
+                fatal "The command needs to be used from the top of the build tree ($ancestor)"
```

Comment:
I was expecting to see automatically changing pwd to top of build tree, but this is actually even better, as there is less room for mistakes :)